airflow/models/baseoperator.py

```diff
@@ -634,6 +651,33 @@ def deps(self) -> Set[BaseTIDep]:
             NotPreviouslySkippedDep(),
         }
 
+    def _set_xcomargs_dependencies(self) -> None:
+        """
+        Resolves upstream dependencies of a task. In this way passing an ``XComArg`
+        as value for a template field will result in creating upstream relation between
+        two tasks.
+
+        **Example**: ::
+
+            with DAG(...):
+                generate_content = GenerateContentOperator(task_id="generate_content")
+                send_email = EmailOperator(..., html_content=generate_content.output)
+
+            # This is equivalent to
+            with DAG(...):
+                generate_content = GenerateContentOperator(task_id="generate_content")
+                send_email = EmailOperator(
+                    ..., html_content="{{ task_instance.xcom_pull('generate_content') }}"
+                )
+                generate_content >> send_email
+
+        """
+        from airflow.models.xcom_arg import XComArg
+        for field in self.template_fields:
+            arg = getattr(self, field)
+            if isinstance(arg, XComArg):
+                self.set_upstream(arg.operator)
```

Comment:
what about pythonOperator? usually, template fields are traversed with `operator.render_template` call, we probably should do the same here

Reply:
> what about pythonOperator? 
What do you mean here @evgenyshulman ?

> usually, template fields are traversed with `operator.render_template` call, we probably should do the same here

I am not sure if I fully understand you here. What exactly is wrong?

Reply:
```
op = PythonOperator( op_kwargs={"a": xarg_value}, op_args=[ xarg_value] )
```
xarg_value will not be resolved into dependencie. Although at .execute xcom values are going to be used automatically

Reply:
Added